cocotb/handle.py

```diff
@@ -83,6 +83,9 @@ def __init__(self, handle, path):
 
         self._name = simulator.get_name_string(self._handle)
         self._type = simulator.get_type_string(self._handle)
+        self._is_port = simulator.get_is_port(self._handle)
+        self._port_direction = simulator.get_port_direction(self._handle)
+        self._port_direction_string = simulator.get_port_direction_string(self._handle)
```

Comment:
This seems like it should be handled by a subclass, not boolean flags.

Reply:
You mean ``class PortObject(NonHierarchyObject):``?
(If I understand correctly what a ``NonHierarchyObject`` is.)